packages/compass-query-bar/src/stores/query-bar-reducer.ts

```diff
@@ -215,19 +246,24 @@ export const changeSchemaFields = (
 type ApplyQueryAction = {
   type: QueryBarActions.ApplyQuery;
   query: unknown;
+  fields: QueryBarState['fields'];
 };
 
 export const applyQuery = (): QueryBarThunkAction<
   false | Record<string, unknown>,
   ApplyQueryAction
 > => {
   return (dispatch, getState, { localAppRegistry }) => {
-    const { fields } = getState();
+    const { fields: _fields } = getState();
+
+    const fields = cloneDeep(_fields);
```

Comment:
By deep cloning the whole state you will cause re-rendering the whole query bar, including all the editors even though in practice only one field might be changed in this action and even this doesn't happen in all cases. Would be nice if we could be more specific about this update